multiversx_sdk_cli/transactions.py

```diff
@@ -132,7 +133,15 @@ def send_and_wait_for_result(transaction: ITransaction, proxy: INetworkProvider,
 def _send_transaction_and_wait_for_result(proxy: INetworkProvider, payload: ITransaction, num_seconds_timeout: int = 100) -> ITransactionOnNetwork:
     AWAIT_TRANSACTION_PERIOD = 5
 
-    tx_hash = proxy.send_transaction(payload)
+    try:
+        tx_hash = proxy.send_transaction(payload)
+    except GenericError as ge:
+        url = ge.url
+        message = ge.data["error"]
+        data = ge.data["data"]
+        code = ge.data["code"]
+        raise ProxyError(message, url, data, code)
```

Comment:
Since we have this in several places now, perhaps use the decorator pattern, and do a custom proxy provider e.g. `CustomNetworkProvider(ProxyNetworkProvider)` that overrides this function, catches & re-throws? 

Alternatively, catch `multiversx_sdk_network_providers.GenericError` in `cli.py`, in the main function, and handle it there?